### PR TITLE
Issue/22 specify requests for middleware

### LINKED ISSUE #22: Activate middleware only for specific requests that require it
As the middleware is active for all requests, this will raise errors for those requests done before assigning an authentification token.
It could be refactored as a function / decorator

### PR DESCRIPTION
**Pull request by mottegk**
Add a "middleware" which can be used for specific routes

Closes #22

A test route has been added to test this feature.
Fixes a bug in the refresh route.

### CODE REVIEW

File: core/services/user_service/controllers/auth_controller.py [hunk 1]
```diff
@@ -48,18 +48,20 @@ def login(email: str, password: str) -> Tuple[str, str]:
...
-    payload = verify_token(refresh_token)
-
-    if not payload:
+    payload_refresh = verify_token(refresh_token)
+    if not payload_refresh:
         raise HTTPException(status_code=401, detail="Invalid refresh token")
 
-    email: str = payload.get("email")
+    # using old access token to get user email
+    payload_access = verify_token(access_token)
```
Comment (use-to) [inline core/services/user_service/controllers/auth_controller.py:62]: Rename

File: core/services/user_service/views/user.py [hunk 2]
```diff
@@ -42,12 +47,25 @@ async def login(data: LoginRequest):
 @router.post("/refresh", response_model=RefreshAccessResponse)
 async def refresh_access(data: RefreshAccessRequest):
     try:
-        access_token, refresh_token = auth_controller.refresh_access_token(
-            data.refresh_token
-        )
+        access_token, refresh_token = \
```
Comment (use-to) [inline core/services/user_service/views/user.py:50]: Maybe call it `old_access_token` or smth?

**Review by use-to — CHANGES_REQUESTED** [review]

File: core/services/user_service/controllers/auth_controller.py (hunk 1 above)
Comment (mottegk) [reply core/services/user_service/controllers/auth_controller.py]: I will implement both a decode_token, which returns the payload and a verify_token which returns a boolean

File: core/services/user_service/views/user.py (hunk 2 above)
Comment (mottegk) [reply core/services/user_service/views/user.py]: This occurence is actually the new_access_token. I'm changing it to new_access_token and old_access_token for better readability.

**Review by use-to — APPROVED** [review]

**Pull request merged**